### ISSUE TITLE
Scope element for steps in item-submission.xml does not work as described in the documentation

### ISSUE THREAD

**Issue by S-Eiberger** [DSpace/DSpace#7957]
**Describe the bug**
**Dspace7**
The scope element for steps in item-submission.xml does not work as described / expected:
https://wiki.lyrasis.org/display/DSDOC7x/Submission+User+Interface#SubmissionUserInterface-Structureofthe%3Cstep-definition%3Etag

**To Reproduce**
Modify the step traditionalpageone as follows:
`<step-definition id="traditionalpageone">
<heading>submit.progressbar.describe.stepone</heading>
<processing-class>org.dspace.app.rest.submit.step.DescribeStep</processing-class>
<type>submission-form</type>
<scope visibilityOutside="read-only">submission</scope>
</step-definition>`

https://github.com/DSpace/DSpace/blob/main/dspace/config/item-submission.xml#L87-L91

This is the same scope configuration as for the license step, where it works as expected (license step is visible and editable during submission, read-only during workflow). However, it did not work for the DescribeStep above. Instead, the step was visible and editable in both contexts (submission and workflow).

Another example (of several visibility configurations tested):

`<step-definition id="traditionalpageone"> 
<heading>submit.progressbar.describe.stepone</heading>
<processing-class>org.dspace.app.rest.submit.step.DescribeStep</processing-class>
<type>submission-form</type>
<scope>submission</scope>
</step-definition>`

was visible in both contexts 

**Expected behavior**
The scope and visibility elements should have the described effects:
https://wiki.lyrasis.org/display/DSDOC7x/Submission+User+Interface#SubmissionUserInterface-Structureofthe%3Cstep-definition%3Etag

**Comment by MW3000** [comment]
I can confirm the same problem. For us it is crucial to be able to limit some steps to scope _workflow_. Is it possible to consider this bug as high priority for 7.4?

**Comment by tdonohue** [comment]
@MW3000 : It's already on our board for 7.4.  I'll bump up the priority to `high` though as it sounds like this is impacting more users than I originally thought.  At this point in time though, it's still waiting for a volunteer to help tackle, so I cannot guarantee what release it'll be in (but a high priority label will raise its visibility)

**Comment by buithaihai** [comment]
Our analysis reveals that this particular bug is due to Angular not respecting the step config it received, specifically: The data is already sent from REST in request `GET /server/api/workflow/workflowitems/<id>?projection=full` that is called by angular, which contains the `SubmissionStepConfig` in question:
```json
            {
              "id" : "traditionalpageone",
              "header" : "submit.progressbar.describe.stepone",
              "mandatory" : true,
              "sectionType" : "submission-form",
              "visibility" : {
                "main" : null,
                "other" : "READONLY"
              },
              "type" : "submissionsection",
              "_links" : {
                [...]
              }
            }
```

Which is similar to that of "license" step. However, only "license" seems to work. Further testing shows that applying ` <scope visibilityOutside="read-only">submission</scope>` to other configs also do nothing. Even further, via chrome debugging, we notice that for "dspace-angular", this code is being used only in "section-license.component.ts":

```typescript
      this.sectionService.isSectionReadOnly(
      ...
```
[Link to source](https://github.com/DSpace/dspace-angular/blob/d2fa8cda6ac2c0139ebf9182140eda45192a1898/src/app/submission/sections/license/section-license.component.ts#L147)

This suggests that other sections are not performing this check and applying the required restrictions accordingly. Therefore it is safe to conclude that changes only need to be made on Angular's side. We would be interested in working on that at a later date when we have more capacity to work with Angular in general.

**Extra thought**: Would it make sense if we add some forms of check server-side? It is curious that there is a check for edit metadata here but it does not account for our use case:
https://github.com/DSpace/DSpace/blob/7b720d65219b8693c58453de0e1b55c0b6f31a5d/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/WorkflowItemRestRepository.java#L265
Our analysis also yields no results on any checks for our use case elsewhere, either. Therefore, we would like to propose an expansion of the scope of this problem to include the necessary checks in the RestAPI, with test coverage if it is considered necessary to do so.

**Comment by buithaihai** [comment]
@tdonohue pinging you just in case

**Comment by tdonohue** [comment]
@buithaihai : It sounds like your recommendation is that this needs fixing in the Angular UI and that you might be interested in "working on that at a later date".  So, I don't see anything needing my immediate feedback?   Since we are finishing up 7.6, it sounds like this won't be ready in time anyways so we'll need to reschedule.

**Comment by tdonohue** [comment]
This appears to be fixed in #2297 in 7.6

Documentation was also updated in https://wiki.lyrasis.org/display/DSDOC7x/Submission+User+Interface#SubmissionUserInterface-CompositionofaField

You can see the summary of my testing that PR here https://github.com/DSpace/dspace-angular/pull/2297#pullrequestreview-1484156875

Closing this as fixed as the description of this ticket matches with what was tested in that PR.  If we find issues still exist, we can always reopen

**Comment by f-rapp** [comment]
Hi, we noticed that what buithaihai wrote about <scope visibilityOutside="read-only">submission</scope> still seems to be an issue with 7.6. He wrote:

"Which is similar to that of "license" step. However, only "license" seems to work. Further testing shows that applying  <scope visibilityOutside="read-only">submission</scope> to other configs also do nothing. "

We tried the config <scope visibilityOutside="read-only">submission</scope> with the upload step, however during workflow the file is still editable etc.
Should we reopen this issue?
